src/java/org/apache/cassandra/concurrent/ResizableThreadPool.java

```diff
@@ -15,22 +15,28 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.cassandra.concurrent;
 
-import java.util.concurrent.BlockingQueue;
-import java.util.concurrent.TimeUnit;
+package org.apache.cassandra.concurrent;
 
-public class JMXConfigurableThreadPoolExecutor extends JMXEnabledThreadPoolExecutor implements JMXConfigurableThreadPoolExecutorMBean
+public interface ResizableThreadPool
 {
+    /**
+     * Returns maximum pool size of thread pool.
+     */
+    public int getCorePoolSize();
```

Comment:
Most of these methods aren't used.  Should `LocalAwareExecutorService` extend `ResizableThreadPool`?

Reply:
Originally I added it make sure the JMX support for resizing pools was common. It's the same as the interface in LocalAwareExecutorService so it makes sense to use the interface.  Updated.